### PR TITLE
To merge UI/sidebar menu with GUI branch

### PR DESCRIPTION
**Pull request by walrys**
Sidebar menu is Degree Requirements Panel is now functional.
@NatashaKSS do look out for React coding mistakes as I'm still learning React and Javascript, thanks!

### CODE REVIEW

File: imports/ui/components/common/Nestable.jsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  }
+  render(){
+    var isCollapsed = this.state.isCollapsed;
+    var listName = this.props.listName;
+
+    return(
+      <li className="dd-item">
+        <div  className="dd-handle" style={styles.list}
+              onClick={this.toggleCollapse.bind(this)}>
+          <i  className={"fa fa-"+ (isCollapsed ? "plus" : "minus")}
```
Comment (NatashaKSS) [inline imports/ui/components/common/Nestable.jsx]: Clever! Changing the icon with a toggle!

File: imports/ui/components/common/Nestable.jsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+          </i>
+          {listName}
+        </div>
+        <OrderedList items={this.props.items} isCollapsed={isCollapsed}/>
+      </li>
+    );
+  }
+}
+
+class OrderedList extends Nestable{
```
Comment (NatashaKSS) [inline imports/ui/components/common/Nestable.jsx]: Interesting way of structuring if else statements within React components! The flow of logic is clear and I know what gets rendered / not rendered. I think I will try to follow your style when I'm dealing with React components which hold a lot of state information.  👍

File: imports/ui/components/sidebar_menu/SidebarPanel.jsx [hunk 3]
```diff
@@ -0,0 +1,143 @@
+import React from 'react';
+
+export default class SidebarPanel extends React.Component {
```
Comment (NatashaKSS) [inline imports/ui/components/sidebar_menu/SidebarPanel.jsx]: For long repeated JSX elements, you can put each repeated segment into a single React component or into a variable within the render{...} method if the element is small enough. You can then loop through a list of those elements to render instead and just specify props to change their titles!

**Comment by walrys** [comment]
Thank you natasha!

**Pull request merged**